### PR TITLE
Add commitid to product version of binaries

### PR DESCRIPTION
**Pull request by dannyvv**
## Description
Add the commitId to the ProductVersion to make finding matching source code more easy.
Using the `+` part after the semver standard.. This matches what MsBuild.exe does.

### Type of Change
- Bug fix (non-breaking change which fixes an issue)

### Why
Improve debugability by making it easier to map sourcecode to a version

### What
Updates to stamp-version and relevant msbuild scripts

## Screenshots
image

 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/11316)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Version.rc [hunk 1]
```diff
@@ -10,6 +10,10 @@
 #define VER_FILEVERSION_STR XSTRINGIZE(RNW_VERSION)
 #define VER_FILEVERSION RNW_MAJOR,RNW_MINOR,RNW_PATCH,RNW_PKG_VERSION_BUILD
 
+#define VER_PRODUCTVERSION_STR XSTRINGIZE(RNW_VERSION+RNW_COMMITID)
+#define VER_PRODUCTVERSION RNW_MAJOR,RNW_MINOR,RNW_PATCH,RNW_PKG_VERSION_BUILD
```
Comment (mganandraj) [inline vnext/Microsoft.ReactNative/Version.rc:14]: Should there be commit_id at the end of this field value ?

**Review by mganandraj — APPROVED** [review]
:shipit:

File: vnext/Microsoft.ReactNative/Version.rc (hunk 1 above)
Comment (dannyvv) [reply vnext/Microsoft.ReactNative/Version.rc:14]: That is not allowed, it is a 4-digit version number. I opted to create the sibbling of `FILEVERSION` for symetry. I could have kept `VER_FILEVERSION` but that felt inconsistent.

**Review by NikoAri — APPROVED** [review]
:shipit:

**Pull request merged**